app/src/main/java/com/github/se/signify/ui/screens/home/HomeScreen.kt

```diff
@@ -188,67 +185,93 @@ fun CameraFeedbackButton(onClick: () -> Unit = {}) {
  * @param numbOfHeaders The number of headers at the top of the list, allowing for an offset when
  *   scrolling to the selected letter.
  */
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun LetterDictionary(
     scrollState: LazyListState,
     coroutineScope: CoroutineScope,
     numbOfHeaders: Int
 ) {
-  var currentLetterIndex by remember { mutableIntStateOf(0) }
   val letters = ('a'..'z').toList()
+  val pagerState = rememberPagerState(initialPage = 0, pageCount = { letters.size })
 
-  Row(
-      modifier = Modifier.fillMaxWidth().wrapContentHeight().testTag("LetterDictionary"),
-      horizontalArrangement = Arrangement.SpaceBetween,
-      verticalAlignment = Alignment.CenterVertically) {
-        IconButton(
-            onClick = {
-              currentLetterIndex = (currentLetterIndex - 1 + letters.size) % letters.size
-            },
-            modifier = Modifier.testTag("LetterDictionaryBack")) {
-              Icon(
-                  Icons.AutoMirrored.Outlined.ArrowBack,
-                  tint = MaterialTheme.colorScheme.onBackground,
-                  contentDescription = "Back")
-            }
+  Box(modifier = Modifier.fillMaxWidth().wrapContentHeight().testTag("LetterDictionary")) {
+    // Back Arrow at the start
+    IconButton(
+        onClick = {
+          coroutineScope.launch {
+            pagerState.animateScrollToPage(
+                (pagerState.currentPage - 1 + letters.size) % letters.size)
+          }
+        },
```

Comment:
The use of coroutines for smooth scrolling is excellent. However, in the futur we can consider adding a safeguard to handle scenarios where the user swipes too quickly across multiple letters in rapid succession.